File: Source/NativeApi/NativeApi/Api/Sizer/SizerItem.cs
#pragma warning disable
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Alternet.Drawing;

namespace NativeApi.Api
{
    public class SizerItem
    {
        public static IntPtr CreateSizerItem(
            IntPtr window,
            int proportion = 0,
            int flag = 0,
            int border = 0,
            IntPtr userData = default) => default;

        public static IntPtr CreateSizerItem2(IntPtr window, IntPtr sizerFlags) => default;

        public static IntPtr CreateSizerItem3(
            IntPtr sizer,
            int proportion = 0,
            int flag = 0,
            int border = 0,
            IntPtr userData = default) => default;

        public static IntPtr CreateSizerItem4(IntPtr sizer, IntPtr sizerFlags) => default;

        public static IntPtr CreateSizerItem5(
            int width,
            int height,
            int proportion = 0,
            int flag = 0,
            int border = 0,
            IntPtr userData = default) => default;

        public static IntPtr CreateSizerItem6(int width, int height, IntPtr sizerFlags) => default;

        public static IntPtr CreateSizerItem7() => default;

        public static void DeleteWindows(IntPtr handle) { }

        // Enable deleting the SizerItem without destroying the contained sizer.
        public static void DetachSizer(IntPtr handle) { }

        // Enable deleting the SizerItem without resetting the sizer in the
        // contained window.
        public static void DetachWindow(IntPtr handle) { }

        public static SizeI GetSize(IntPtr handle) => default;
        public static SizeI CalcMin(IntPtr handle) => default;
        public static void SetDimension(IntPtr handle, PointI pos, SizeI size) { }

        public static SizeI GetMinSize(IntPtr handle) => default;
        public static SizeI GetMinSizeWithBorder(IntPtr handle) => default;
        public static SizeI GetMaxSize(IntPtr handle) => default;
        public static SizeI GetMaxSizeWithBorder(IntPtr handle) => default;

        public static void SetMinSize(IntPtr handle, int x, int y) { }
        public static void SetInitSize(IntPtr handle, int x, int y) { }

        // if either of dimensions is zero, ratio is assumed to be 1
        // to avoid "divide by zero" errors
        public static void SetRatio(IntPtr handle, int width, int height) { }
        public static void SetRatio2(IntPtr handle, float ratio) { }
        public static float GetRatio(IntPtr handle) => default;

        public static RectI GetRect(IntPtr handle) => default;

        // set a sizer item id (different from a window id, all sizer items,
        // including spacers, can have an associated id)
        public static void SetId(IntPtr handle, int id) { }
        public static int GetId(IntPtr handle) => default;

        public static bool IsWindow(IntPtr handle) => default;
        public static bool IsSizer(IntPtr handle) => default;
        public static bool IsSpacer(IntPtr handle) => default;

        public static void SetProportion(IntPtr handle, int proportion) { }
        public static int GetProportion(IntPtr handle) => default;
        public static void SetFlag(IntPtr handle, int flag) { }
        public static int GetFlag(IntPtr handle) => default;
        public static void SetBorder(IntPtr handle, int border) { }
        public static int GetBorder(IntPtr handle) => default;

        public static IntPtr GetWindow(IntPtr handle) => default;
        public static IntPtr GetSizer(IntPtr handle) => default;
        public static SizeI GetSpacer(IntPtr handle) => default;

        // This function behaves obviously for the windows and spacers but for the
        // sizers it returns true if any sizer element is shown and only returns
        // false if all of them are hidden. Also, it always returns true if
        // wxRESERVE_SPACE_EVEN_IF_HIDDEN flag was used.
        public static bool IsShown(IntPtr handle) => default;

        public static void Show(IntPtr handle, bool show) { }

        public static void SetUserData(IntPtr handle, IntPtr userData) { }
        public static IntPtr GetUserData(IntPtr handle) => default;
        public static PointI GetPosition(IntPtr handle) => default;

        // Called once the first component of an item has been decided. This is
        // used in algorithms that depend on knowing the size in one direction
        // before the min size in the other direction can be known.
        // Returns true if it made use of the information (and min size was changed).
        public static bool InformFirstDirection(IntPtr handle, int direction,
            int size, int availableOtherDir = -1) => default;

        // these functions delete the current contents of the item if it's a sizer
        // or a spacer but not if it is a window
        public static void AssignWindow(IntPtr handle, IntPtr window) { }
        public static void AssignSizer(IntPtr handle, IntPtr sizer) { }
        public static void AssignSpacer(IntPtr handle, int w, int h) { }

    }
}

File: Source/Alternet.UI.Pal/SizerItem.cpp
#include "SizerItem.h"

namespace Alternet::UI
{
    SizerItem::SizerItem()
    {
    }

    SizerItem::~SizerItem()
    {
    }

    void* SizerItem::CreateSizerItem(void* window, int proportion,
        int flag, int border, void* userData)
    {
        return nullptr;
    }

    void* SizerItem::CreateSizerItem2(void* window, void* sizerFlags)
    {
        return nullptr;
    }

    void* SizerItem::CreateSizerItem3(void* sizer, int proportion, int flag, int border, void* userData)
    {
        return nullptr;
    }

    void* SizerItem::CreateSizerItem4(void* sizer, void* sizerFlags)
    {
        return nullptr;
    }

    void* SizerItem::CreateSizerItem5(int width, int height, int proportion, int flag,
        int border, void* userData)
    {
        return nullptr;
    }

    void* SizerItem::CreateSizerItem6(int width, int height, void* sizerFlags)
    {
        return nullptr;
    }

    void* SizerItem::CreateSizerItem7()
    {
        return nullptr;
    }

    void SizerItem::DeleteWindows(void* handle)
    {
    }

    void SizerItem::DetachSizer(void* handle)
    {
    }

    void SizerItem::DetachWindow(void* handle)
    {
    }

    Int32Size SizerItem::GetSize(void* handle)
    {
        return Int32Size();
    }

    Int32Size SizerItem::CalcMin(void* handle)
    {
        return Int32Size();
    }

    void SizerItem::SetDimension(void* handle, const Int32Point& pos, const Int32Size& size)
    {
    }

    Int32Size SizerItem::GetMinSize(void* handle)
    {
        return Int32Size();
    }

    Int32Size SizerItem::GetMinSizeWithBorder(void* handle)
    {
        return Int32Size();
    }

    Int32Size SizerItem::GetMaxSize(void* handle)
    {
        return Int32Size();
    }

    Int32Size SizerItem::GetMaxSizeWithBorder(void* handle)
    {
        return Int32Size();
    }

    void SizerItem::SetMinSize(void* handle, int x, int y)
    {
    }

    void SizerItem::SetInitSize(void* handle, int x, int y)
    {
    }

    void SizerItem::SetRatio(void* handle, int width, int height)
    {
    }

    void SizerItem::SetRatio2(void* handle, float ratio)
    {
    }

    float SizerItem::GetRatio(void* handle)
    {
        return 0;
    }

    Int32Rect SizerItem::GetRect(void* handle)
    {
        return Int32Rect();
    }

    void SizerItem::SetId(void* handle, int id)
    {
    }

    int SizerItem::GetId(void* handle)
    {
        return 0;
    }

    bool SizerItem::IsWindow(void* handle)
    {
        return false;
    }

    bool SizerItem::IsSizer(void* handle)
    {
        return false;
    }

    bool SizerItem::IsSpacer(void* handle)
    {
        return false;
    }

    void SizerItem::SetProportion(void* handle, int proportion)
    {
    }

    int SizerItem::GetProportion(void* handle)
    {
        return 0;
    }

    void SizerItem::SetFlag(void* handle, int flag)
    {
    }

    int SizerItem::GetFlag(void* handle)
    {
        return 0;
    }

    void SizerItem::SetBorder(void* handle, int border)
    {
    }

    int SizerItem::GetBorder(void* handle)
    {
        return 0;
    }

    void* SizerItem::GetWindow(void* handle)
    {
        return nullptr;
    }

    void* SizerItem::GetSizer(void* handle)
    {
        return nullptr;
    }

    Int32Size SizerItem::GetSpacer(void* handle)
    {
        return Int32Size();
    }

    bool SizerItem::IsShown(void* handle)
    {
        return false;
    }

    void SizerItem::Show(void* handle, bool show)
    {
    }

    void SizerItem::SetUserData(void* handle, void* userData)
    {
    }

    void* SizerItem::GetUserData(void* handle)
    {
        return nullptr;
    }

    Int32Point SizerItem::GetPosition(void* handle)
    {
        return Int32Point();
    }

    bool SizerItem::InformFirstDirection(void* handle, int direction, int size, int availableOtherDir)
    {
        return false;
    }

    void SizerItem::AssignWindow(void* handle, void* window)
    {
    }

    void SizerItem::AssignSizer(void* handle, void* sizer)
    {
    }

    void SizerItem::AssignSpacer(void* handle, int w, int h)
    {
    }
}
